docs/topics/auth/customizing.txt

```diff
@@ -259,12 +259,48 @@ Handling object permissions
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Django's permission framework has a foundation for object permissions, though
-there is no implementation for it in the core. That means that checking for
-object permissions will always return ``False`` or an empty list (depending on
-the check performed). An authentication backend will receive the keyword
-parameters ``obj`` and ``user_obj`` for each object related authorization
+there is no implementation for it in the core. An authentication backend will
+receive the keyword parameter ``obj`` for each object related authorization
 method and can return the object level permission as appropriate.
 
+The default :class:`~django.contrib.auth.backends.ModelBackend` only handles
+global permissions. That means that checking for object permissions will always
+return ``False`` or an empty set (depending on the check performed).
+Conversely, it is expected that any backend that handles object permissions
```

Comment:
My idea here was to extend the specific behavior of `ModelBackend` to a genaral rule: Backends should handle either global or object permissions. For authors of global permission backends this means that they should always check `obj is None` (on object permission backends the equivalent `obj is something` needs to be checked anyway, so that is less of an issue).

Any idea where to put this instead? (do you even agree with that rule?)

Reply:
OK. Cor. I'm not sure (at all). 

I think I can easily imagine someone creating a single backend that uses object permissions but then falls back to the general permissions... — I'm not sure I'd want to legislate against that. 

(The permissions system has been ≈unchanged for ≈10 years: IMO one of it's strengths is not having strong opinions — I know people ask for the "superset" behaviour, so OK we can facilitate that but my bias would be towards less, rather than more...)

Reply:
I understand your point about keeping it flexible. I removed this part and restored the original text.